staker-rewards/lib/rewards-calculator.js

```diff
@@ -26,12 +26,25 @@ export default class RewardsCalculator {
     const operatorsRewardsFactors = []
 
     for (const operatorParameters of operatorsParameters) {
-      const { keepStaked, ethTotal } = operatorParameters.operatorAssets
-
-      const requirementsViolations = this.checkRequirementsViolations(
+      let requirementsViolations = this.checkRequirementsViolations(
         operatorParameters
       )
 
+      if (await this.isUndelegatingOperator(operatorParameters.operator)) {
+        if (
+          requirementsViolations.includes("poolRequirementFulfilledAtStart")
+        ) {
+          requirementsViolations = requirementsViolations.filter(
+            (item) => item !== "poolRequirementFulfilledAtStart"
+          )
+        }
+
+        operatorParameters.operatorAssets.ethTotal =
+          operatorParameters.operatorAssets.ethBonded
+      }
```

Comment:
I do not think it's correct as it does not include ETH that was withdrawn. When calculating assets, we do:
https://github.com/keep-network/keep-ecdsa/blob/e5132c4f0235649196af6b5d9fe7ef0bb59c839f/staker-rewards/lib/assets-calculator.js#L58

I even do wonder if the check for undelegating operator shouldn't be done in `assets-calculator.js` as well, so that we keep all the logic about `ethTotal` calculations there.

Reply:
But withdraws are about the unbonded value. I don't think we should subtract the unbonded value withdrawals from the bonded value sum. Am I right?

Reply:
If we move this to the `assets-calculator.js` we will have some duplications because we have to check undelegations in two places:
- in the `assets-calculator` to do `ethTotal = ethBonded`
- in the `rewards-calculator` to remove the sortition pool condition violation if present

If it is acceptable we can do it.

Reply:
> But withdraws are about the unbonded value. I don't think we should subtract the unbonded value withdrawals from the bonded value sum. Am I right?

Correct, though bonded could become unbonded and then withdrawn during the interval. That's why we always need to subtract.


> If we move this to the `assets-calculator.js` we will have some duplications because we have to check undelegations in two places

I think this is acceptable. We'll have to repeat `isUndelegating` check so there is a cost of an additional request but the structure will be cleaner. To me, `assets-calculator.js` reflects *Operator Parameters* section in the readme while `rewards-calculator.js` validates everything in *Requirements* section.